### PR TITLE
Sms get

### PR DESCRIPTION
**Pull request by natebrennand**
Adds a GET method that takes in a SID and gets the resource.
Utilizes a helper method in common/ that can be reused for others.

has a program in the examples dir

### CODE REVIEW

**Comment by natebrennand** [comment]
waiting on more tests before merging

**Comment by natebrennand** [comment]
ready to be merged pending approval

File: common/response.go [hunk 1]
```diff
@@ -1,6 +1,7 @@
 package common
 
 import (
+	// "fmt"
```
Comment (chelpu) [inline common/response.go:4]: you can probably delete this

File: sms/callback.go [hunk 2]
```diff
@@ -14,7 +14,8 @@ type Media struct {
 }
 
 // Represents the callback sent everytime the status of the message is updated.
-// Visit https://www.twilio.com/docs/api/rest/sending-messages#status-callback-parameter for more detaiils
+//
+// Visit https://www.twilio.com/docs/api/rest/sending-messages#status-callback-parameter for more details
```
Comment (chelpu) [inline sms/callback.go:18]: niiice

File: sms/callback.go [hunk 3]
```diff
@@ -43,34 +44,15 @@ func (cb *Callback) Parse(req *http.Request) error {
...
-			Location:   msgLocation,
-		},
+		MessageSid:      req.PostFormValue("MessageSid"),
+		SmsSid:          req.PostFormValue("SmsSid"),
+		Body:            req.PostFormValue("Body"),
+		NumMedia:        numMedia,
+		MessageStatus:   req.PostFormValue("MessageStatus"),
+		ErrorCode:       req.PostFormValue("ErrorCode"),
+		MediaList:       mediaArray,
+		StandardRequest: common.ParseStandardRequest(req),
```
Comment (chelpu) [inline sms/callback.go:55]: this is awesome

File: sms/callback.go (hunk 2 above)
Comment (natebrennand) [inline sms/callback.go:18]: it renders better in the godoc

File: common/response.go (hunk 1 above)
Comment (natebrennand) [inline common/response.go:4]: will remove in next PR

File: sms/callback.go (hunk 3 above)
Comment (natebrennand) [inline sms/callback.go:55]: D.R.Y.

**Pull request merged**